music21/harmony.py

```diff
@@ -1945,7 +1949,7 @@ def _updatePitches(self):
         ['C3', 'E-3', 'G3']
 
         >>> [str(pi) for pi in CS(root='C', bass='B', kind='major-ninth').pitches]
-        ['B2', 'C3', 'D3', 'E3', 'G3']
+        ['B1', 'D2', 'C3', 'E3', 'G3']
```

Comment:
B1 is pretty low -- is this because we take C3 as a basis but why is D below C here.  I'm a little confused.

Reply:
Culprit is this:

https://github.com/cuthbertLab/music21/blob/ed2049ef2897ffa69c73be45b329fc22fac6fec6/music21/harmony.py#L2033-L2038

Going into that block, the pitches are C3, E3, G3, B3, D4. The block bumps C, E, G to octave 5, then quits, instead of skipping the B and then acting on the D to bump the D to D6.

Then, the block that begins `while self._hasPitchAboveC4` brings everything down two octaves.

The change in this diff is that now bass notes are given a default octave of 3 just like roots. Before the root default octave was 3 and the bass octave was silent so we defaulted it from pitch.py.

That meant that this block of code acts on the D4, which it won't in the current version (because D4 > B4):

https://github.com/cuthbertLab/music21/blob/ed2049ef2897ffa69c73be45b329fc22fac6fec6/music21/harmony.py#L2043-L2050

This second block seems duplicative of the first block. I'm going to push another commit for that first block that de-fangs it a little bit.